### PR TITLE
[#1597] feat(core): Support entity decoder

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

- Add decoder method to decode byte array to name identifier
- Add some detailed information if `NonEmptyEntityException` occurs

### Why are the changes needed?

Make the log more clear and easy to debug.
Fix: #1597 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

UTs

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java [hunk 1]
```diff
@@ -380,10 +380,15 @@ public boolean delete(NameIdentifier ident, EntityType entityType, boolean casca
...
+                    for (Pair<byte[], byte[]> pair : kvs) {
+                      subEntities.add(entityKeyEncoder.decode(pair.getLeft()).getLeft());
+                    }
+
                     throw new NonEmptyEntityException(
                         String.format(
-                            "Entity %s has sub-entities, you should remove sub-entities first",
-                            ident));
+                            "Entity %s has sub-entities %s, you should remove sub-entities first",
+                            ident, subEntities));
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java:391]: Will this `subEntities` be so large when there're so many entities? If so, is it necessary to print them all?

Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java:391]: No, I have limited the data retrieved from Gravitino, please see:

```java
  new KvRangeScan.KvRangeScanBuilder()
                              .start(directChild)
                              .end(endKey)
                              .startInclusive(true)
                              .endInclusive(false)
                              .limit(1)
                              .build());
```

**Review by diqiu50 — APPROVED** [review]
LGTM

File: core/src/test/java/com/datastrato/gravitino/storage/kv/TestEntityKeyEncoding.java [hunk 2]
```diff
@@ -262,4 +272,50 @@ public void testNamespaceEncoding()
                   EntityType.COLUMN));
     }
   }
+
+  @Test
+  void testSpecialCharacterDecoder() throws IOException {
```
Comment (jerryshao) [inline core/src/test/java/com/datastrato/gravitino/storage/kv/TestEntityKeyEncoding.java:277]: I would suggest to use parameterized test to test more cases?

Comment (yuqi1129) [reply core/src/test/java/com/datastrato/gravitino/storage/kv/TestEntityKeyEncoding.java]: Fixed

**Review by jerryshao — APPROVED** [review]

**Pull request merged**